hip-config.cmake.in

```diff
@@ -51,7 +51,9 @@ set_and_check(hip_HIPCC_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipcc")
 set_and_check(hip_HIPCONFIG_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipconfig")
 
 if(HIP_COMPILER STREQUAL "clang")
-  if(CMAKE_CXX_COMPILER MATCHES ".*hipcc")
+  if(HIP_CXX_COMPILER)
+    set(HIP_CLANG_ROOT ${HIP_CXX_COMPILER})
```

Comment:
This wont set the HIP_CLANG_ROOT correctly if `HIP_CXX_COMPILER` is set to /opt/rocm/llvm/bin/clang++.